src/framework/theme/components/form-field/form-field.component.ts

```diff
@@ -0,0 +1,219 @@
+}
+
+/*
+ * NbFormFieldComponent
+ *
+ * @styles
+ *
+ * form-field-addon-basic-text-color:
+ * form-field-addon-basic-highlight-text-color:
+ * form-field-addon-primary-text-color:
+ * form-field-addon-primary-highlight-text-color:
+ * form-field-addon-success-text-color:
+ * form-field-addon-success-highlight-text-color:
+ * form-field-addon-info-text-color:
+ * form-field-addon-info-highlight-text-color:
+ * form-field-addon-warning-text-color:
+ * form-field-addon-warning-highlight-text-color:
+ * form-field-addon-danger-text-color:
+ * form-field-addon-danger-highlight-text-color:
+ * form-field-addon-control-text-color:
+ * form-field-addon-control-highlight-text-color:
+ * form-field-addon-disabled-text-color:
+ * form-field-addon-tiny-height:
+ * form-field-addon-tiny-width:
+ * form-field-addon-tiny-icon-size:
+ * form-field-addon-tiny-font-size:
+ * form-field-addon-tiny-line-height:
+ * form-field-addon-tiny-font-weight:
+ * form-field-addon-small-height:
+ * form-field-addon-small-width:
+ * form-field-addon-small-icon-size:
+ * form-field-addon-small-font-size:
+ * form-field-addon-small-line-height:
+ * form-field-addon-small-font-weight:
+ * form-field-addon-medium-height:
+ * form-field-addon-medium-width:
+ * form-field-addon-medium-icon-size:
+ * form-field-addon-medium-font-size:
+ * form-field-addon-medium-line-height:
+ * form-field-addon-medium-font-weight:
+ * form-field-addon-large-height:
+ * form-field-addon-large-width:
+ * form-field-addon-large-icon-size:
+ * form-field-addon-large-font-size:
+ * form-field-addon-large-line-height:
+ * form-field-addon-large-font-weight:
+ * form-field-addon-giant-height:
+ * form-field-addon-giant-width:
+ * form-field-addon-giant-icon-size:
+ * form-field-addon-giant-font-size:
+ * form-field-addon-giant-line-height:
+ * form-field-addon-giant-font-weight:
+ **/
+@Component({
+  selector: 'nb-form-field',
+  styleUrls: ['./form-field.component.scss'],
+  templateUrl: './form-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NbFormFieldComponent implements AfterContentChecked, AfterContentInit, OnDestroy {
+
+  protected readonly destroy$ = new Subject<void>();
+
+  protected formControlState$ = new BehaviorSubject<NbFormControlState>({
+    status: 'basic',
+    size: 'medium',
+    focused: false,
+    disabled: false,
+  });
+  prefixClasses$: Observable<string[]> = this.getAddonClassesObservable('prefix');
+  suffixClasses$: Observable<string[]> = this.getAddonClassesObservable('suffix');
+
+  @ContentChildren(NbPrefixDirective, { descendants: true }) prefix: QueryList<NbPrefixDirective>;
+  @ContentChildren(NbSuffixDirective, { descendants: true }) suffix: QueryList<NbSuffixDirective>;
+
+  @ContentChild(NbInputDirective, { static: false }) formControl: NbInputDirective;
+  @ContentChild(NbInputDirective, { static: false, read: ElementRef }) formControlElement: ElementRef<HTMLInputElement>;
+
```

Comment:
I would personally prefer to name it `input` instead of `formControl` since we're querying input